components/datetime/src/fields/symbols.rs

```diff
@@ -147,8 +147,8 @@ impl From<FieldSymbol> for char {
             FieldSymbol::Hour(hour) => match hour {
                 Hour::H11 => 'K',
                 Hour::H12 => 'h',
-                Hour::H23 => 'H',
-                Hour::H24 => 'k',
+                Hour::H23 => 'k',
+                Hour::H24 => 'H',
```

Comment:
Reviewing: According to UTS 35:

https://unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table

h = Hour [1-12] = h12
H = Hour [0-23] = h23
K = Hour [0-11] = h11
k = Hour [1-24] = h24
j = locale preference
J = locale preference, omit day period

So, wasn't it correct before this change?

Reply:
Yeah, I screwed this up in this PR. I got confused that h12 and h23 are what are stored in skeletons, and then misread the symbol information. Thanks for double checking this. I'll try and sort it out in my other PR.

Reply:
The `h` `H` being `h12` ans `h23`, and then `K` being `h11` and `k` being `h24` is inconsistent in what symbols match that my brain refuses to parse things correctly.